exotica_core/src/kinematic_tree.cpp

```diff
@@ -1042,14 +1116,16 @@ void KinematicTree::ResetJointLimits()
         {
             auto& ControlledJoint = controlled_joints_map_.at(vars[i]);
             int index = ControlledJoint.lock()->control_id;
-            controlled_joints_[index].lock()->joint_limits = {model_->getVariableBounds(vars[i]).min_position_, model_->getVariableBounds(vars[i]).max_position_};
+
+            // Last element should be model_->getVariableBounds(vars[i]).max_acceleration_, but the URDF-parser
+            // in ros_control seems to skip over parsing acceleration limits (https://github.com/ros-controls/ros_control/issues/350#issuecomment-411670354)
```

Comment:
Important: ROS-Control != ROS. The joint_limits_interface is ROS Control. The URDF specification does not know of an acceleration limit:
https://github.com/ros/urdfdom/blob/0da4b20675cdbe14b532d484a1c17df85b4e1584/urdf_parser/src/joint.cpp#L97-L166

Suggestion: Add `effort_limit` and `velocity_limit`. Do not instantiate acceleration limits by parsing since they are not part of the standard. If you want to have acceleration limits, use the getter/setter - and a boolean flag `has_acceleration_limits` which defaults to `false`.

Reply:
I agree, let's go with Wolf's suggestion. It's worth the extra work.